recursion.rb

```diff
@@ -1,17 +1,82 @@
+# Lisa Rolczynski
 
+# RC: n! = n * (n-1)!
+# BC: 1! = 1
+# BC: 0! = 1
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+
+  return n * fact(n-1)
 end
 
+# BC: fib(1) = 1
+# BC: fib(2) = 1
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+
+  fib(n-1) + fib(n-2)
 end
 
+# palindromes
+# BC: if there's 1 or 0 letters left, return true
 def pal(s)
+  s_array = s.chars
+  start_index = 0
+  end_index = s_array.length - 1
+
+  pal_helper(s_array, start_index, end_index)
 end
 
+def pal_helper(s_array, start_index, end_index)
+  return true if end_index - start_index <= 1
+
+  if s_array[start_index] == s_array[end_index]
+    return true && pal_helper(s_array, start_index+1, end_index-1)
+  else
+    return false
+  end
+end
+
+
 def binary(n)
+  binary_helper(n,"","")
 end
 
+def binary_helper(digits,binary,result)
+  if digits == 0
+    result += binary + " "
+    return result
+  else
+    result = binary_helper(digits-1,binary+"0",result)
+    result = binary_helper(digits-1,binary+"1",result)
+  end
+
+  return result
+end
+
+
 def travel(x,y)
+  travel_helper(x,y)
+end
+
+def travel_helper(x,y,directions="",result="")
+  if x == 0 && y == 0
+    result += directions + " "
+    return result
+  elsif x == 0
+    return result = travel_helper(x,y-1,directions+"N",result)
+  elsif y == 0
+    return result = travel_helper(x-1,y,directions+"E",result)
+  else
+    result = travel_helper(x-1,y,directions+"E",result)
+    result = travel_helper(x,y-1,directions+"N",result)
+  end
+
+  return result
```

Comment:
This looks good! I think you can eliminate the "return = " in front of the x == 0 and y == 0 cases since those statements are in elsif's and there's a return at the end of the method